### PR TITLE
chore!: bump sdk fork to v1.17.0

### PR DESCRIPTION
**Pull request by cmwaters**
Bumps to the latest version of the SDK. Note that this deactivates the use of the EVM address. To fulfill that functionality we will need to reintroduce a mapping in https://github.com/celestiaorg/celestia-app/pull/2169

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Review by SweeXordious — COMMENTED** [review]
[not blocking]
This is super cool, but as long as the mapping PR is not merged, the QGB will be broken, therefore main will be broken.
Does it make sense to postpone bumping until we have that PR reviewed and ready to be merge? so that we don't have a list of commits which all contain a broken QGB in between

**Comment by cmwaters** [comment]
> [not blocking] This is super cool, but as long as the mapping PR is not merged, the QGB will be broken, therefore main will be broken. Does it make sense to postpone bumping until we have that PR reviewed and ready to be merge? so that we don't have a list of commits which all contain a broken QGB in between

I was hoping I could kind of do these as two separate commits. Maybe it makes sense to merge the other one first

File: x/qgb/abci_test.go [hunk 1]
```diff
@@ -83,22 +82,20 @@ func TestValsetCreationWhenEditingEVMAddr(t *testing.T) {
...
 	)
-	_, err = msgServer.EditValidator(ctx, editMsg)
+	_, err := msgServer.EditValidator(ctx, editMsg)
 	require.NoError(t, err)
 	staking.EndBlocker(ctx, input.StakingKeeper)
 	qgb.EndBlocker(ctx, *pk)
 	ctx = ctx.WithBlockHeight(ctx.BlockHeight() + 10)
 
-	assert.Equal(t, currentAttestationNonce+1, pk.GetLatestAttestationNonce(ctx))
+	// FIXME: this needs to change to 2 once we have a proper implementation of editing the EVM address
```
Comment (rootulp) [inline x/qgb/abci_test.go:97]: [question] not sure what you mean by "2"

**Review by rootulp — APPROVED** [review]
I think this needs a make proto-gen

**Comment by SweeXordious** [comment]
> I was hoping I could kind of do these as two separate commits

Yes, as separate commit, but better they are sequential. Like we merge this one and the next commit on main would be the mapping commit.
This would keep main not breaking the qgb as low as possible

File: x/qgb/abci_test.go (hunk 1 above)
Comment (cmwaters) [reply x/qgb/abci_test.go:97]: Ah it's currently 1 but we should be asserting that the latest nonce is 2

Comment (cmwaters) [reply x/qgb/abci_test.go:97]: I also should have @-ed myself but forgot to

**Review by evan-forbes — APPROVED** [review]
this LGTM

fwiw I think its fine to merge to main and but then just not backport it yet to v1.x. This avoids using some temporaray feature branch, which is arguably just as bad as leaving main in a temporary unreleasable state (a property I'm not sure we need to preserve any more since we never want to cut releases directly from main)

**Review by rootulp — APPROVED** [review]

**Pull request merged**